feat: add noir tests for events

nventuro opened the pull request:
This started out as me wanting to remove the commitment from an event message, as it is unused, and then realizing that testing this was quite sad because we had no infra to do it.

We now do.

I did not publicly expose it yet as there are some open questions (what to do if we get too many events? how do we handle events emitted by contracts?), but those are more API questions and not related to the internals. With this we can at least use the capabilities ourselves for internal testing.

mverzilli commented on yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts:
```diff
@@ -755,6 +825,8 @@ export class PXEOracleInterface implements ExecutionDataProvider {
       throw new Error(`Could not find tx effect for tx hash ${txHash} as of block number ${syncedBlockNumber}`);
     }
 
+    this.log.debug('delivering event', { txNullifiers: txEffect.data.nullifiers });
```
question more for my own understanding: do we have any criteria for producing these logs other than "this might be useful"?

nventuro replied:
This was me doing printf debugging :sweat_smile:

mverzilli commented on yarn-project/txe/src/rpc_translator.ts:
```diff
@@ -267,6 +276,39 @@ export class RPCTranslator {
+      throw new Error(`Array of length ${events.length} larger than maxLen ${MAX_PRIVATE_EVENTS_PER_TXE_QUERY}`);
+    }
+
+    if (events.some(e => e.length > MAX_EVENT_LEN)) {
+      throw new Error(`Some private event has length larger than maxLen ${MAX_EVENT_LEN}`);
+    }
```
wouldn't it be worth saying which events are breaking this invariant?

nventuro replied:
I didn't expand as I expect for this workaround to be short lived, and I'm not even sure it's possible to deliver such a large event - it's only there for completeness.

mverzilli reviewed (approved):
Very nice!

nventuro committed:
feat: add noir tests for events

This started out as me wanting to remove the commitment from an event message, as it is unused, and then realizing that testing this was quite sad because we had no infra to do it.

We now do.

I did not publicly expose it yet as there are some open questions (what to do if we get too many events? how do we handle events emitted by contracts?), but those are more API questions and not related to the internals. With this we can at least use the capabilities ourselves for internal testing.

Co-authored-by: Nicolás Venturo <nicolas.venturo@gmail.com>
